org.eclipse.debug.core/core/org/eclipse/debug/internal/core/LaunchManager.java

```diff
@@ -530,20 +530,27 @@ public static Document getDocument() throws ParserConfigurationException {
 	 * platform line separators.
 	 *
 	 * @param doc document to serialize
+	 * @param lineDelimiter the new line separator to use
 	 * @return the document as a string
 	 * @throws TransformerException if an unrecoverable error occurs during the
 	 *             serialization
 	 * @throws IOException if I/O error occurs
 	 */
-	public static String serializeDocument(Document doc) throws TransformerException, IOException {
+	public static String serializeDocument(Document doc, String lineDelimiter) throws TransformerException, IOException {
```

Comment:
This is the crust of the change.The existing transformer-based API does not provide a way to control the newline character.  Using the LS serializer allows you to control it.

Reply:
does it change any formating?

Reply:
I set the options to make this backward compatible. Running "cat <file.launch>" did not yield any difference other than the newline ending. There was a change in the presence of the `<?xml version="1.0" encoding="UTF-8"?>` but that was fixed by setting the "isStandalone" property to true (the new line nuymber 545 below). So I tried to keep the before/after identical and it seems to have worked.